ARCs/arc-0004.md

```diff
@@ -143,12 +143,31 @@ interface Method {
     /** Optional, user-friendly description for the argument */
     desc?: string
   }>,
+  /** A list of allowed OnCompletion values for the method. Must not be empty */
+  allowedActions: OnCompletionAction[],
```

Comment:
If these are specified in an interface, must a contract support all of them in order to be considered an implementation of the interface?  Can it support more?  I think perhaps this belongs only in Contract methods, not Interface methods.  I dislike that nearly so many methods will be annotated that they allow optin.  I prefer that the ABI says "all methods should support optin".  I prefer to make rules that try to regularize ABI contracts, rather than expand they ABI to support describing arbitrary complexity.